agent/consul/txn_endpoint.go

```diff
@@ -81,9 +81,9 @@ func (t *Txn) preCheck(authorizer acl.Authorizer, ops structs.TxnOps) structs.Tx
 			}
 
 			service := &op.Service.Service
-			// This is intentionally nil as we will authorize the request
-			// using vetServiceTxnOp next instead of doing it in servicePreApply
-			if err := servicePreApply(service, nil); err != nil {
+			// acl.ManageAll is used here because the request will be authorized
+			// later using vetServiceTxnOp.
+			if err := servicePreApply(service, acl.ManageAll()); err != nil {
```

Comment:
Calling out this change as it's slightly different from the rest.

The goal of this chain of PRs is to remove these nil checks, so we replace the `nil` we pass in here with `ManageAll` to achieve the same goal.